DNN Platform/Library/Common/Globals.cs

```diff
@@ -2534,14 +2534,7 @@ public static string AddHTTP(string strURL)
             {
                 if (strURL.IndexOf("mailto:") == -1 && strURL.IndexOf("://") == -1 && strURL.IndexOf("~") == -1 && strURL.IndexOf("\\\\") == -1)
                 {
-                    if (HttpContext.Current != null && HttpContext.Current.Request.IsSecureConnection)
-                    {
-                        strURL = "https://" + strURL;
-                    }
-                    else
-                    {
-                        strURL = "http://" + strURL;
-                    }
+                    strURL = UrlUtils.IsSecureConnectionOrSslOffload(HttpContext.Current.Request) ? "https://" : "http://" + strURL;

```

Comment:
it should need brackets on the ?: block like:
strURL = (UrlUtils.IsSecureConnectionOrSslOffload(HttpContext.Current.Request) ? "https://" : "http://") + strURL;